airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -273,6 +291,22 @@ def create_connector_report(results: List[StepResult]) -> ConnectorReport:
                 metadata_upload_results = await metadata_upload_step.run()
                 results.append(metadata_upload_results)
 
+            # Try to convert the context to a PyPIPublishContext. If it returns None, it means we don't need to publish to pypi.
+            pypi_context = await PyPIPublishContext.from_connector_context(context)
```

Comment:
💅 Lets ensure we move this logic to a named function to keep `run_connector_publish_pipeline` readable and focused on coordination